prow/plugins/plugins.go

```diff
@@ -138,9 +135,41 @@ type PluginClient struct {
 	Logger *logrus.Entry
 }
 
-type PluginAgent struct {
-	PluginClient
+func NewAgent(configAgent *config.Agent, pluginConfigAgent *ConfigAgent, clientAgent *ClientAgent, logger *logrus.Entry) Agent {
+	return Agent{
+		GitHubClient: clientAgent.GitHubClient,
+		KubeClient:   clientAgent.KubeClient,
+		GitClient:    clientAgent.GitClient,
+		SlackClient:  clientAgent.SlackClient,
+		Config:       configAgent.Config(),
+		PluginConfig: pluginConfigAgent.Config(),
+		Logger:       logger,
+	}
+}
+
+func (a *Agent) OwnersClient() *repoowners.Client {
+	return repoowners.NewClient(
+		a.GitClient, a.GitHubClient,
+		a.Config, a.PluginConfig.MDYAMLEnabled,
+		a.PluginConfig.SkipCollaborators,
+	)
+}
+
+func (a *Agent) CommentPruner(org, repo string, pr int) *commentpruner.EventClient {
+	return commentpruner.NewEventClient(
```

Comment:
Using `commentpruner` like this defeats its purpose. There should be at most 1 event client for every GitHub event. All handlers of the event should share the same event client in order to get the caching benefits.

Reply:
They are ... there's nothing special about this client, it just wraps the (single, shared) GitHub client

Reply:
Ah, the client's internal comment cache ... hm.

Reply:
See here: https://github.com/kubernetes/test-infra/blob/d351abbaf62a0aa8e107d93a8c1febb654c95d0a/prow/commentpruner/commentpruner.go#L34-L51
The problem is that `CommentPruner` is being called per handler instead of per event, meaning  the consumers of the event are not sharing the client and instead each have their own (defeating the caching).